src/Bundler.js

```diff
@@ -318,8 +329,9 @@ class Bundler extends EventEmitter {
         this.hmr.emitError(err);
       }
 
-      if (process.env.NODE_ENV === 'production') {
-        process.exitCode = 1;
+      if (process.env.NODE_ENV === 'production' || !initialised) {
+        await this.stop();
+        process.exit(1);
```

Comment:
Apparently `process.exitCode = 1;` didn't exit the process anymore for some reason, so changed this to be a lil more aggresive and added `await this.stop();` to make sure servers and workers shutdown properly